### PR TITLE
Phonehome

### PR DESCRIPTION
**Pull request by cristi-iacob**

### CODE REVIEW

File: src/phonehome.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#ifndef PHONEHOME_H
+#define PHONEHOME_H
+#include "src/pulseview/pv/devices/device.hpp"
+#include "preferences.h"
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QTextCodec>
+
+namespace adiscope {
+class PhoneHome: public QObject
```
Comment (dNechita) [inline src/phonehome.h]: Why singleton?

File: src/tool_launcher.cpp [hunk 2]
```diff
@@ -780,6 +798,29 @@ void ToolLauncher::setupHomepage()
 
 	QWidget *homepage = new QWidget(ui->stackedWidget);
 	QVBoxLayout *layout = new QVBoxLayout(homepage);
+
+	connect(PhoneHome::getInstance(), &PhoneHome::scopyVersionChanged, this, [=] () {
+		QLabel* versionLabel = new QLabel();
+
+		if (PhoneHome::getScopyVersion().toStdString().compare("") == 0) {
```
Comment (DanielGuramulta) [inline src/tool_launcher.cpp]: Maybe use `PhoneHome::getScopyVersion().isEmpty()`, if I understand correctly what this is doing, it's checking if the version `QString` is empty.

File: src/tool_launcher.cpp [hunk 3]
```diff
@@ -780,6 +798,29 @@ void ToolLauncher::setupHomepage()
...
+			versionLabel->setStyleSheet("{ color : white; }");
+			versionLabel->setText("Unable to check for latest Scopy version!");
+		} else if (PhoneHome::getScopyVersion().toStdString().compare(std::string("v") + std::string(PROJECT_VERSION)) != 0) {
```
Comment (DanielGuramulta) [inline src/tool_launcher.cpp]: Also here, I think `QString` has operator `>` or `!=` overloaded so just
```cpp
PhoneHome::getScopyVersion() <operator> QString("v" + QString(PROJECT_VERSION))
```

File: src/phonehome.cpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+void PhoneHome::setPreferences(Preferences* preferences)
+{
+	PhoneHome::preferences = preferences;
+}
+
+void PhoneHome::versionsRequest()
+{
+	QNetworkAccessManager* manager = new QNetworkAccessManager();
+	connect(manager, &QNetworkAccessManager::finished, PhoneHome::getInstance(), &PhoneHome::onVersionsRequestFinished);
+	connect(manager, SIGNAL(finished(QNetworkReply*)), manager, SLOT(deleteLater()));
```
[the remark this answers is not in the document]
Comment (cristi-iacob) [reply src/phonehome.cpp]: Added the changes and force pushed the branch.

File: src/info_page.hpp [hunk 5]
```diff
@@ -27,6 +27,7 @@
 #include <QFuture>
 
 #include "iio.h"
+#include "phonehome.h"
```
[the remark this answers is not in the document]
Comment (cristi-iacob) [reply src/info_page.hpp]: Solved.

File: src/info_page.cpp [hunk 6]
```diff
@@ -438,3 +451,15 @@ void M2kInfoPage::blinkTimeout()
 				    "done_led_overwrite_powerdown",
 				    !oldVal);
 }
+
+int M2kInfoPage::checkLatestFwVersion(QString currentVersion) {
```
[the remark this answers is not in the document]
Comment (cristi-iacob) [reply src/info_page.cpp]: Solved.

File: src/preferences.h [hunk 7]
```diff
@@ -95,6 +95,15 @@ class Preferences : public QWidget
 	bool getDigital_decoders_enabled() const;
 	void setDigital_decoders_enabled(bool value);
 
+	bool getAutomatical_version_checking_enabled() const;
+	void setAutomatical_version_checking_enabled(bool value);
+
+	QString getCheck_updates_url() const;
+	void setCheck_update_url(QString link);
```
[the remark this answers is not in the document]
Comment (cristi-iacob) [reply src/preferences.h]: Solved.

File: src/tool_launcher.hpp [hunk 8]
```diff
@@ -32,6 +32,7 @@
 #include <QStringList>
 #include <info_widget.h>
 #include <QTextBrowser>
+#include <QMessageBox>
```
[the remark this answers is not in the document]
Comment (cristi-iacob) [reply src/tool_launcher.hpp]: Solved.

File: src/tool_launcher.cpp (hunk 2 above)
Comment (cristi-iacob) [reply src/tool_launcher.cpp]: Solved.

File: src/tool_launcher.cpp (hunk 3 above)
Comment (cristi-iacob) [reply src/tool_launcher.cpp]: Solved.

**Comment by adisuciu** [comment]
I rebased and added some more functionality in this PR https://github.com/analogdevicesinc/scopy/pull/975

closing this

**Pull request closed without merging**